### PR TITLE
feat(container.signature): add image digest to ContainerSignatureValidationService API output

### PR DESCRIPTION
**Pull request by mattdibi**
While working on #5129 I realised we didn't have a easy way to extract the image digest from a remote image (i.e. an image that was not already pulled on the system). I thought about expanding thee `container.orchestration` APIs to this end (see #5131) but the juice doesn't seem to be worth the squeeze. Even more so realising that all the tools I tested for the container signature validation already return the image digest as output

Therefore this PR updates the `ContainerSignatureValidationService` APIs so that, upon verifying the signature, it returns also the image digest. This can be later used to reference the verified image in a unique and robust way.

Merging this PR will impact #5126 and #5129 and avoid the need for #5131 entirely.

**Cosign**

```
cosign-linux-amd64 verify --key cosign.pub mattiadibieth/cosign-test:latest  --insecure-ignore-tlog

WARNING: Skipping tlog verification is an insecure practice that lacks of transparency and auditability verification for the signature.

Verification for index.docker.io/mattiadibieth/cosign-test:latest --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - The signatures were verified against the specified public key
[
  {
    "critical": {
      "identity": {
        "docker-reference": "index.docker.io/mattiadibieth/cosign-test"
      },
      "image": {
        "docker-manifest-digest": "sha256:da53a44247cdffda84f979b73eb2064a75ec46378a0a53d4df1e064269842324"
      },
      "type": "cosign container image signature"
    },
    "optional": null
  }
]
```

**Notary V1**

```
docker trust inspect mattiadibieth/notary-test:latest
[
    {
        "Name": "mattiadibieth/notary-test:latest",
        "SignedTags": [
            {
                "SignedTag": "latest",
                "Digest": "da53a44247cdffda84f979b73eb2064a75ec46378a0a53d4df1e064269842324",
                "Signers": [
                    "Repo Admin"
                ]
            }
        ],
        "Signers": [],
        "AdministrativeKeys": [
            {
                "Name": "Root",
                "Keys": [
                    {
                        "ID": "2561662fba4201f4e7bb7226b7ad669f8c09bd5366215acd47c906c8ae137a4b"
                    }
                ]
            },
            {
                "Name": "Repository",
                "Keys": [
                    {
                        "ID": "ae3ccd530d445fc78458d73e0b3a32e1a3e99bce72c67da479fcbc004ec4bcb5"
                    }
                ]
            }
        ]
    }
]
```

**Notary V2**

```
notation verify $IMAGE
Successfully verified signature for localhost:5001/notation-test@sha256:da53a44247cdffda84f979b73eb2064a75ec46378a0a53d4df1e064269842324
```

---

**Additional details**

To perform this same task by our own we should:

- Find out the correct registry
- Find out the correct authentication endpoint for that registry and request an auth token
- Request the manifest to the registry using the aforementioned token
- Parse the manifest retrieving the digest for the image compatible with our architecture

Nothing difficult per se but finding out the correct registry/auth endpoint doesn't seem to be straightforward.

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Class representing the result of the signature validation performed by {@link:ContainerSignatureValidationService}
+ * 
+ * The result of the validation is composed of two main parts: whether or not the container image signature was
+ * validated and the container image digest (in the "shaXXX:YYY" format).
+ *
+ * @since 2.7
+ */
+public final class ValidationResult {
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java:32]:
```suggestion
@ProviderType
public final class ValidationResult {
```

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package org.eclipse.kura.container.signature;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java:13]: Copyright header missing

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Class representing the result of the signature validation performed by {@link:ContainerSignatureValidationService}
+ * 
+ * The result of the validation is composed of two main parts: whether or not the container image signature was
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java]:
```suggestion
 * The validation result is composed of two main parts: whether or not the container image signature was
```

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    public ValidationResult() {
+        // Nothing to do
+    }
+
+    public ValidationResult(boolean signatureValid, String digest) {
+        this.imageDigest = Optional.of(Objects.requireNonNull(digest));
+        this.isSignatureValid = Objects.requireNonNull(signatureValid);
+
+        if (this.isSignatureValid && (!this.imageDigest.isPresent() || this.imageDigest.get().isEmpty())) {
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java]: Does it make sense to have it in line 30 before the assignments?

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java (hunk 2 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java:13]: DAMMIT! One day I will open a PR with all the copyright headers first try I promise

Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java:13]: Addressed in c5c668e11cc023457cfb0ed269a2ce508504fd51

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java (hunk 4 above)
Comment (mattdibi) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java]: Addressed in 8ff5490112d50df1415e7dfd1bb526a0baa247fb

**Comment by mattdibi** [comment]
@MMaiero Sonar is whining about code coverage but I can't find where to put the unit tests it wants (we don't have a `org.eclipse.kura.api.test` package in our tests). Some coverage will be addressed in #5126 (as soon as this PR is merged), should I create some unit just for `ValidationResults` anyway? 🤔

**Review by MMaiero — APPROVED** [review]

**Comment by MMaiero** [comment]
Let's keep it as is.
We don't have tests for the APIs

**Comment by mattdibi** [comment]
Pipeline fails due to low coverage
image

Merging anyway as per our agreement.

**Pull request merged**